cmd/system-probe/api/module/loader_linux.go

```diff
@@ -12,11 +12,17 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/ebpf"
 )
 
-func preRegister(_ *config.Config) error {
+func preRegister(cfg *config.Config) error {
+	if !cfg.RequiresEBPF() {
```

Comment:
I'd invert this, so you only do the ebpf setup if ebpf is required. That way any non-ebpf setup could go below.